server/flask-app/urls/api.py

```diff
@@ -0,0 +1,117 @@
+from flask import Blueprint, jsonify, request, redirect
+from flask_jwt_extended import jwt_required, get_jwt_identity
+from flasgger import swag_from
+
+from models import Alert, User, DosageHistory, SensorReading, Sensor, db
+
+api = Blueprint('api', __name__)
+
+ # Get the user ID from the JWT token
+def get_user_by_jwt():
+    user_id = get_jwt_identity()
+    user = User.query.get(user_id)
+    return user
+
+#Redirect from homepage to docs
+@api.route('/')
+def docs_redirect():
+    return redirect('/apidocs')
+
+@api.errorhandler(404)
+def page_not_found(e):
+    return jsonify({"error": "Page not found"}), 404
+
+@api.route('/user-devices', methods=['GET'])
+@jwt_required()
+@swag_from('../swagger_templates/get_user_devices.yml')
+def get_user_devices():
+    user = get_user_by_jwt()
+    if not user:
+        return jsonify({'message': 'User not found.'}), 400
+
+    devices = [{'device_id': d.device_id, 'name': d.name} for d in user.devices]
+    return jsonify(devices=devices), 200
+
+@api.route('/sensors/<int:device_id>', methods=['GET'])
+@jwt_required()
+@swag_from('../swagger_templates/get_user_sensors.yml')
+def get_user_sensors(device_id):
+    user = get_user_by_jwt()
+    if not user:
+        return jsonify({'message': 'User not found.'}), 404
+
+    sensors = Sensor.query.filter_by(device_id=device_id).all()
+
+    sensors_list = [{'sensor_id': s.sensor_id} for s in sensors]
+    return jsonify(sensors=sensors_list), 200
```

Comment:
Możemy tutaj zwracać wszystkie parametry sensora? Plus zwracałbym tutaj ostatni odczyt z tego sensora bo to później w apce bedzie potrzebne

Reply:
Zastanawiałem się nad zwróceniem wszystkich parametrów sensora i mogę to dodać w sumie. Co do zwracania ostatniego odczytu to dodam query do endpointa z sensor_readings, który zamiast całej listy pokaże tylko ten ostatni odczyt. Dam znać jak dodam te rzeczy